SONARJAVA-5284 Create rule S7178: Injecting data into static fields is not supported by Spring (#4611)
SonarSource/rspec: rules/S7178/java/rule.adoc

== Why is this an issue?

Spring dependency injection framework does not support injecting data into static fields. When @Value, @Inject, or @Autowired are applied to static fields, they are ignored.

> What is the potential impact?

Reviewer: ```suggestion
=== What is the potential impact?
```

Reviewer 2: :+1:

* Null Values: Uninitialized static fields annotated with @Value, @Inject, or @Autowired will not be initialized by Spring, potentially causing NullPointerException at runtime.
* Confusing Code: The presence of injection annotations on static fields can mislead developers into believing that the fields will be populated by Spring.

> This rule raises an issue when a static will is annotated with @Value, @Inject, or @Autowired.

Reviewer: ```suggestion
This rule raises an issue when a static field is annotated with @Value, @Inject, or @Autowired.
```

Reviewer 2: nice catch :+1:

== How to fix it

Either use an instance field instead of a static field or remove the @Value, @Inject, or @Autowired annotation and initialize the field.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
@Component
public class MyComponent {

    @Value("${my.app.prop}")

>     private static SomeDependency dependency; // Noncompliant, @Value will be ignored and no value will be injected

Reviewer: ```suggestion
    private static SomeDependency dependency; // Non compliant, @Value will be ignored and no value will be injected
```

Reviewer 2: :+1:
